dlpy/tests/test_image_captioning.py

```diff
@@ -0,0 +1,409 @@
+                                                                obj_detect_model=detection_model,
+                                                                word_embeddings_file=word_embeddings,
+                                                                objects_width=4000))
+
+    # works
+    def test_ImageCaptioning_1(self):
+        if self.data_dir is None:
+            unittest.TestCase.skipTest(self, "DLPY_DATA_DIR is not set in the environment variables")
+
+        model = ImageCaptioning(self.s,rnn_type='GRU')
+
+        model = ImageCaptioning(self.s,rnn_type='RNN')
+
+        model = ImageCaptioning(self.s)
+
+        model.print_summary()
+
+        self.assertTrue(model is not None)
+
+    def test_ImageCaptioning_2(self):
+        if self.data_dir is None:
+            unittest.TestCase.skipTest(self,"DLPY_DATA_DIR is not set in the environment variables")
+
+        model = ImageCaptioning(self.s,num_blocks=1)
+
+        model = ImageCaptioning(self.s,num_blocks=10)
+
+        model.print_summary()
+
+        self.assertTrue(model is not None)
+
+    def test_results_to_dict_1(self):
+        if self.data_dir is None:
+            unittest.TestCase.skipTest(self, "DLPY_DATA_DIR is not set in the environment variables")
+        new_dict = dict(_DL_Pred_=['predict1','predict2','predict3'],caption=['capt1','capt2','capt3'],_filename_0=['file1','file2','file3'])
+        results = CASTable.from_dict(self.s,new_dict)
+
+        self.assertTrue(scored_results_to_dict(results) is not None)
+
+    def test_results_to_dict_2(self):
+        if self.data_dir is None:
+            unittest.TestCase.skipTest(self, "DLPY_DATA_DIR is not set in the environment variables")
+
+        result_tbl = self.s.CASTable('not_a_table')
+        self.assertRaises(SWATError,lambda:
+                                    scored_results_to_dict(result_tbl))
+
+    def test_capt_len_1(self):
+        if self.data_dir is None:
+            unittest.TestCase.skipTest(self, "DLPY_DATA_DIR is not set in the environment variables")
+        captions_file = self.data_dir + 'image_captions.txt'
+        self.assertEquals(get_max_capt_len(captions_file,delimiter='\t'),21)
+
+    #     wrong delimiter
+    def test_capt_len_2(self):
+        if self.data_dir is None:
+            unittest.TestCase.skipTest(self, "DLPY_DATA_DIR is not set in the environment variables")
+        captions_file = self.data_dir + 'image_captions.txt'
+        self.assertRaises(DLPyError,lambda:
+                          get_max_capt_len(captions_file,delimiter=','))
+
+    # bad file
+    def test_capt_len_3(self):
+        if self.data_dir is None:
+            unittest.TestCase.skipTest(self, "DLPY_DATA_DIR is not set in the environment variables")
+
+        captions_file = "doesnt/exist.txt"
+        self.assertRaises(FileNotFoundError,lambda:
+                          get_max_capt_len(captions_file,delimiter='\t'))
```

Comment:
I just want to thank you for paying attention to unit tests. great work.

Reply:
please do not forget to anonymize your jupyter notebook. no username, local path, machine names. I can help you with that - let me know if you need help. thanks

Reply:
lets' dont repeat the same files on two locations. one under tests/datasources should be fine.